fluent_pose_synthesis/data/create_data.py

```diff
@@ -0,0 +1,340 @@
+            print(f"Total signs: {total_signs}")
+            print(f"Replaced signs: {replaced_signs}")
+
+
+def save_pose_to_file(pose_object: Pose, file_path: Path):
+    """
+    Saves a Pose object to a file.
+    Args:
+        pose_object (Pose): The Pose object to save.
+        file_path (Path): The file path to save the pose.
+    """
+    with open(file_path, "wb") as file:
+        pose_object.write(file)
+
+
+def setup_logging(output_dir: Path, log_filename: str="processing.log") -> Path:
+    """Setup logging configuration."""
+    log_file_path = Path(output_dir) / log_filename
+    logger = logging.getLogger()
+    logger.handlers.clear()
+    logger.setLevel(logging.INFO)
+
+    file_handler = logging.FileHandler(log_file_path)
+    file_handler.setFormatter(logging.Formatter("%(asctime)s - %(levelname)s - %(message)s"))
+    logger.addHandler(file_handler)
+
+    console_handler = logging.StreamHandler()
+    console_handler.setFormatter(logging.Formatter("%(asctime)s - %(levelname)s - %(message)s"))
+    logger.addHandler(console_handler)
+    
+    return log_file_path
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Process and save pose dataset.")
+    parser.add_argument("--corpus_dir", type=str, required=True, help="Path to the DGS Corpus directory. If the dataset is not downloaded, it will be downloaded to this directory.")
+    parser.add_argument("--dictionary_dir", type=str, required=True, help="Path to the DGS Types dictionary directory. If the dataset is not downloaded, it will be downloaded to this directory.")
+    parser.add_argument("--output_dir", type=str, required=True, help="Output directory to save processed pose files.")
+    parser.add_argument("--max_examples", type=int, default=None, help="Maximum number of examples to process.")
+    args = parser.parse_args()
+
+    # Initialize the DGSDataset class
+    dgs_pose_dataset = DGSPoseDataset(
+        corpus_dir=Path(args.corpus_dir),
+        dictionary_dir=Path(args.dictionary_dir),
+        max_examples=args.max_examples
+    )
+    # Ensure output directory exists
+    output_path = Path(args.output_dir)
+    os.makedirs(output_path, exist_ok=True)
+    # Setup logging
+    log_file_path = setup_logging(output_path)
+    logging.info(f"Logging to file: {log_file_path}")
+    # Global counter to track total processed examples
+    global_counter = {"processed": 0}
+
+    # Iterate through the dataset splits
+    for split in ["train", "validation", "test"]:
+        split_output_path = output_path / split
+        os.makedirs(split_output_path, exist_ok=True)
+        logging.info(f"Starting processing for split: {split}")
+        # Process and save the dataset
+        for idx, processed_sentence in enumerate(dgs_pose_dataset.generate_dataset(split=split, global_counter=global_counter)):
+            # Save original pose
+            original_pose_path = split_output_path / f"{split}_{idx + 1}_original.pose"
```

Comment:
this looks pretty good. I would love it if we also saved the metadata in a file (perhaps a JSON) such that we have:
`train_1.original.pose`, `train_1.updated.pose` and `train_1.metadata.json` or something like that, if we would like to debug